src/uct/api/uct_def.h

```diff
@@ -79,6 +79,46 @@ typedef struct uct_ep_addr       uct_ep_addr_t;
  * @}
  */
 
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Structure for scatter-gather I/O.
+ *
+ * Specifies a list of buffers which can be used within a single data transfer
+ * function call.
+ *
+   @verbatim
+    buffer
+    |
+    +---------+-------+---------+-------+---------+-------+
+    | Payload | empty | Payload | empty | Payload | empty |
+    +---------+-------+---------+-------+---------+-------+
+    <--------------------- length ----------------------->
+    <---- stride ---->
+   @endverbatim
+ *
+ * @note If the @a length is zero, the @a buffer will not be touched.
+ *       If the @a length greater than zero, the @a buffer must contain valid
+ *       pointer and @a count must be greater than zero.
+ *
+ * @note If the @a count is not greater than one, the @a length will be used as
+ *       the size of single data transfer.
+ *       If the @a count greater than one, the @a stride must contain a size in
+ *       bytes that represented distance between beginnings of consecutive
+ *       payload data. In this case, a single data transfer size will be the
+ *       minimum between @a length and the product of @a stride and @a count.
+ *
```

Comment:
@shamisp I would propose to replace `count` variable (because it could calculated by `length` / `stride`) to `payload_len` to provide size of the payload.
If the `stride` bigger than zero we should treat the `buffer` as consecutive set of payloads with `stride` (as distance) and `payload_len` as the size in bytes.